README.md

```diff
@@ -35,25 +89,31 @@ specify the following types yourself:
 * `int8_t` = signed 8bit integer
 * `uint8_t` = unsigned 8bit integer
 
+In addition to that you will need to specify `NULL`. For both we have a commented
+template where just need to fill in your system specific values.
+
 Make sure you specify the whole range (64bit sized integers are used on a 32bit
 system for some sensor data).
 
-### sensirion\_hw\_i2c\_implementation.c
-Use this file when using a dedicated I2C controller (use software I2C from the
-next section for bit banging on GPIOs.)
-
-Use this file to implement the I2C HAL via the I2C controller of your
-system. Implement the documented function stubs: an init, a read, a write and a
-delay function. For further information please check out some of the sample
-implementations in the `i2c/hw/sample-implementations` folder in this
-repository.
-
-### sensirion\_sw\_i2c\_implementation.c
-Use this file if you don't have a dedicated hardware I2C controller (otherwise
-see the previous section) to emulate an I2C controller in software by
-bit-banging GPIOs. You will thus need to implement the stubs necessary to drive
-the GPIO pins used for SDA (data line) and SCL (clock line) along with the delay
-function, that must be precise enough to drive the pins at the desired
-frequency.  For further information please check out some of the sample
-implementations in the `i2c/sw/sample-implementations` folder in this
-repository.
+### sensirion\_i2c\_hal.c
+
+As already explained earlier is all the I2C related code in this repository based
+on Sensirion's I2C HAL. It is defined in `sensirion_i2c_hal.h` and implemented in
+the corresponding `.c` file, both located in the `i2c/` folder. To use the I2c part
+of `embedded-common` you either need to implement the `sensirion_i2c_hal.c` file or
+replace the unimplemented version in `i2c/` with the correct sample implementation
+from `i2c/sample-implementation/`.
```

Comment:
I found this sentence a bit confusing: Maybe better: "you either need to implement the functions in `sensirion_i2c_hal.c` or replace it with a sample implementation from `i2c/sample-implementation/`".